src/dmd/backend/bcomplex.d

```diff
@@ -24,28 +22,291 @@ struct Complex_f
+                w = sqrt(x) * sqrt(0.5 * (1 + sqrt(1 + r * r)));
+            }
+            else
+            {
+                r = x / y;
+                w = sqrt(y) * sqrt(0.5 * (r + sqrt(1 + r * r)));
+            }
+            if (z.re >= 0)
+            {
+                c.re = w;
+                c.im = z.im / (w + w);
+            }
+            else
+            {
+                c.im = (z.im >= 0) ? w : -w;
+                c.re = z.im / (c.im + c.im);
+            }
+        }
+        return c;
+    }
 }
 
 struct Complex_d
 {
     double re, im;
 
-    static Complex_d div(ref Complex_d x, ref Complex_d y);
-    static Complex_d mul(ref Complex_d x, ref Complex_d y);
-    static targ_ldouble abs(ref Complex_d z);
-    static Complex_d sqrtc(ref Complex_d z);
+    static Complex_d div(ref Complex_d x, ref Complex_d y)
+    {
+        Complex_d q;
+        real r;
+        real den;
+
+        if (fabs(y.re) < fabs(y.im))
+        {
+            r = y.re / y.im;
+            den = y.im + r * y.re;
+            q.re = (x.re * r + x.im) / den;
+            q.im = (x.im * r - x.re) / den;
+        }
+        else
+        {
+            r = y.im / y.re;
+            den = y.re + r * y.im;
+            q.re = (x.re + r * x.im) / den;
+            q.im = (x.im - r * x.re) / den;
+        }
+        return q;
+    }
+
+    static Complex_d mul(ref Complex_d x, ref Complex_d y)
+    {
+        Complex_d p;
+        p.re = x.re * y.re - x.im * y.im;
+        p.im = x.im * y.re + x.re * y.im;
+        return p;
+    }
+
+    static real abs(ref Complex_d z)
+    {
+        real x,y,ans,temp;
+        x = fabs(z.re);
+        y = fabs(z.im);
+        if (x == 0)
+            ans = y;
+        else if (y == 0)
+            ans = x;
+        else if (x > y)
+        {
+            temp = y / x;
+            ans = x * sqrt(1 + temp * temp);
+        }
+        else
+        {
+            temp = x / y;
+            ans = y * sqrt(1 + temp * temp);
+        }
+        return ans;
+    }
+
+    static Complex_d sqrtc(ref Complex_d z)
+    {
+        Complex_d c;
+        real x,y,w,r;
+
+        if (z.re == 0 && z.im == 0)
+        {
+            c.re = 0;
+            c.im = 0;
+        }
+        else
+        {
+            x = fabs(z.re);
+            y = fabs(z.im);
+            if (x >= y)
+            {
+                r = y / x;
+                w = sqrt(x) * sqrt(0.5 * (1 + sqrt(1 + r * r)));
+            }
+            else
+            {
+                r = x / y;
+                w = sqrt(y) * sqrt(0.5 * (r + sqrt(1 + r * r)));
+            }
+            if (z.re >= 0)
+            {
+                c.re = w;
+                c.im = z.im / (w + w);
+            }
+            else
+            {
+                c.im = (z.im >= 0) ? w : -w;
+                c.re = z.im / (2 * c.im);
+            }
+        }
+        return c;
+    }
 }
 
+
 struct Complex_ld
 {
-    targ_ldouble re, im;
+    real re, im;
```

Comment:
You have to keep `targ_ldouble` here because when built with LDC on Windows, `real` is the same as `double`.

Reply:
ok